ffDesign_AutoFillet.py

```diff
@@ -0,0 +1,133 @@
+from PySide import QtCore, QtGui
+
+import FreeCADGui as Gui
+import FreeCAD as App
+import Part
+
+import ffDesign_Utils as Utils
+
+
+def make_auto_fillets(
+    body,
+    *,
+    axis: str = "X",
+    radius: App.Units.Quantity,
+):
+    Utils.assert_body(body)
+
+    radius = App.Units.Quantity(radius)
+    assert radius.Unit.Type == "Length"
+
+    tip = body.Tip
+    if tip is None:
+        raise Utils.ffDesignError("Active body has no tip feature!")
+
+    def check_tangent_vertical(tangent) -> bool:
+        if axis == "X":
+            return abs(tangent.x) >= 0.5
+        elif axis == "Y":
+            return abs(tangent.y) >= 0.5
+        elif axis == "Z":
+            return abs(tangent.z) >= 0.5
+        else:
+            raise Utils.ffDesignError(f"Unknown axis {axis!r} (must be X, Y, or Z)")
+
+    filtered_edges = []
+    for idx, edge in enumerate(tip.Shape.Edges, start=1):
+        param = edge.ParameterRange
+        Utils.Log.info(f"Parameter range for Edge{idx}: {param!r}")
+        tangent_start = edge.tangentAt(param[0])
+        tangent_mid = edge.tangentAt((param[0] + param[1]) / 2)
+        tangent_end = edge.tangentAt(param[1])
+
+        # We accept edges where all three tangents are more than 45°
+        if (
+            check_tangent_vertical(tangent_start)
+            and check_tangent_vertical(tangent_mid)
+            and check_tangent_vertical(tangent_end)
+        ):
+            filtered_edges.append(f"Edge{idx}")
+
+    if filtered_edges == []:
+        raise Utils.ffDesignError("Did not find any vertical edges in the active body!")
+
+    fillet = body.newObject("PartDesign::Fillet", "VerticalFillets")
```

Comment:
was my setting the new tip for the active body explicitly unnecessary and it's actually done automatically?